File: src/scrt/com/packet/ACData.java
/*******************************************************************************
 * Copyright (C) 2017-2018 César Benito Lamata
 * 
 * This file is part of SCRT.
 * 
 * SCRT is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 * 
 * SCRT is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with SCRT.  If not, see <http://www.gnu.org/licenses/>.
 ******************************************************************************/
package scrt.com.packet;

import java.io.IOException;
import java.io.InputStream;
import java.util.ArrayList;
import java.util.List;

import scrt.Orientation;

public class ACData extends StatePacket implements DataPacket
{
	public Orientation dir = Orientation.None;
	public ACData(ACID id)
	{
		super(id);
	}
	@Override
	public List<Integer> getListState()
	{
		List<Integer> data = new ArrayList<>();
		data.addAll(id.getId());
		data.add(dir.ordinal());
		return data;
	}
	public static ACData byState(InputStream i) throws IOException
	{
		i.read();
		ACData ac = new ACData(new ACID(i));
		ac.dir = Orientation.values()[i.read()];
		return ac;
	}
}

File: common/packet/ACData.cpp
#include "ACData.h"
ACData::ACData(shared_ptr<ACID> id) : StatePacket(id, PacketType::ACData, PacketGroup::Data) {}
vector<char> ACData::getListState()
{
    vector<char> data = id->getId();
    data.push_back((char)dir);
    return data;
}
Packet *ACData::byState(char *&data)
{
    ACData *ac = new ACData(shared_ptr<ACID>(new ACID(data)));
    ac->dir = (Orientation)*data++;
    return ac;
}